tests/functional/z_cluster/test_scale_mon_pods.py

```diff
@@ -0,0 +1,322 @@
+import logging
+import pytest
+import time
+
+from ocs_ci.ocs import ocp, constants
+from ocs_ci.framework.pytest_customization.marks import (
+    brown_squad,
+    post_ocs_upgrade,
+    skipif_less_than_five_workers,
+)
+from ocs_ci.framework.testlib import (
+    ManageTest,
+    skipif_ocs_version,
+)
+from ocs_ci.framework import config
+from ocs_ci.ocs.cluster import CephCluster
+from ocs_ci.utility import prometheus
+from ocs_ci.ocs.exceptions import ResourceNotFoundError, CommandFailed
+from ocs_ci.ocs.node import (
+    get_worker_nodes,
+    get_node_objs,
+    drain_nodes,
+    wait_for_nodes_status,
+    schedule_nodes,
+)
+from ocs_ci.ocs.resources.pod import verify_mon_pod_running
+from ocs_ci.ocs.cluster import is_vsphere_ipi_cluster
+from ocs_ci.utility.retry import retry
+from ocs_ci.ocs.resources import pod
+
+log = logging.getLogger(__name__)
+
+
+@brown_squad
+@skipif_less_than_five_workers
+@skipif_ocs_version("<4.15")
+class TestFiveMonInCluster(ManageTest):
+    def __init__(self):
+
+        """
+        Initializer function
+
+        """
+
+        self.mon_count = 5
+        self.ceph_cluster = CephCluster()
+
+        self.storagecluster_obj = ocp.OCP(
+            resource_name=constants.DEFAULT_CLUSTERNAME,
+            namespace=config.ENV_DATA["cluster_namespace"],
+            kind=constants.STORAGECLUSTER,
+        )
+
+    def assign_dummy_racks(self):
+        """
+        Assign node labels to given nodes based on given rack lists.
+
+        """
+        overwrite = True
+        if len(self.nodes) % len(self.racks) != 0:
+            msg = "number of nodes is not divisible by number of racks"
+            log.error(msg)
+            raise ValueError(msg)
+        node_h = ocp.OCP(kind="node")
+        for node, rack in zip(self.nodes, self.racks):
+            log.info("labeling node %s with %s=%s", node, constants.RACK_LABEL, rack)
+            oc_cmd = f"label node {node} {constants.RACK_LABEL}={rack}"
+            if overwrite:
+                oc_cmd += " --overwrite"
+            node_h.exec_oc_cmd(command=oc_cmd)
+
+    def are_rack_labels_present(self):
+        """
+        Check that there are no nodes without rack labels.
+
+        Returns:
+            Bool: True if all nodes have a rack label, False otherwise.
+        """
+        node_h = ocp.OCP(kind="node")
+        nodes_labeled = node_h.get(selector=constants.RACK_LABEL)
+        node_names = [n["metadata"]["name"] for n in nodes_labeled["items"]]
+        log.info("nodes with '%s' rack label: %s", constants.RACK_LABEL, node_names)
+        return len(nodes_labeled["items"]) == len(get_worker_nodes())
```

Comment:
Can this function moved to some library file?

Reply:
These functions are test specific as we are increasing failure domain count by assigning dummy racks. Moving it to library file might create more confusion.